metaflow/plugins/airflow/airflow.py

```diff
@@ -172,63 +172,40 @@ def _get_retry_delay(self, node):
-        seen = set()
         airflow_params = []
         type_transform_dict = {
             int.__name__: "integer",
             str.__name__: "string",
             bool.__name__: "string",
             float.__name__: "number",
-            JSONTypeClass.name: "string",
-            FilePathClass.name: "string",
-        }
-
-        type_parser = {
-            bool.__name__: lambda v: str(v),
         }
 
         for var, param in self.flow._get_parameters():
-            # Throw an exception if the parameter is specified twice.
-            norm = param.name.lower()
-            if norm in seen:
-                raise MetaflowException(
-                    "Parameter *%s* is specified twice. "
-                    "Note that parameter names are "
-                    "case-insensitive." % param.name
-                )
-            seen.add(norm)
-
             # Airflow requires defaults set for parameters.
-            if "default" not in param.kwargs:
-                raise MetaflowException(
-                    "Parameter *%s* does not have a "
-                    "default value. "
-                    "A default value is required for parameters when deploying flows on Airflow."
-                )
             value = deploy_time_eval(param.kwargs.get("default"))
             # Setting airflow related param args.
-            param_type = param.kwargs.get("type")
             airflow_param = dict(
                 name=param.name,
             )
             if value is not None:
                 airflow_param["default"] = value
             if param.kwargs.get("help"):
                 airflow_param["description"] = param.kwargs.get("help")
-            if param_type is not None:
-                # Todo (fast-follow): Check if we can support more `click.Param` types
-                param_type_name = getattr(param_type, "__name__", None)
-                if not param_type_name and isinstance(
-                    param_type, (JSONTypeClass, FilePathClass)
-                ):
-                    # `JSONTypeClass`/`FilePathClass` have no __name__ attribute so
-                    # we need to explicitly check if `param_type` is an instance of those
-                    param_type_name = param_type.name
-
-                if param_type_name in type_transform_dict:
-                    airflow_param["type"] = type_transform_dict[param_type_name]
-                    if param_type_name in type_parser and value is not None:
-                        airflow_param["default"] = type_parser[param_type_name](value)
+
+            # Since we will always have a default value and `deploy_time_eval` resolved that to an actual value
+            # we can just use the `default` to infer the object's type.
+            # This avoids parsing/indentifying types like `JSONType` or `FilePathClass`
+            # which are returned by calling `param.kwargs.get("type")`
+            param_type = type(airflow_param["default"])
```

Comment:
Does JSONType return a string or can it return a dict? I suspect it returns the dict which may require a dumps at least so maybe something that does a dumps if it is not in param_type?

Reply:
Isn't the default value of JSONType if supposed to be a JSON string?

Reply:
My assumption is that JSONType's default value should be a string.

Reply:
I don't want to run json.dumps because _I don't know_ if it may mess up parsing when the user has already run json.dumps when feeding the default.

Reply:
Our [Docs](http://docs.metaflow.org/metaflow/basics#how-to-define-parameters-for-flows) expect JSON strings

Reply:
Yes, and I verified the evaluation of the default will return a string and not actually do the loads; only the conversion function does the loads so this is all good. Sorry for the confusion.